.github/workflows/continuous-integration.yml

```diff
@@ -71,6 +71,18 @@ jobs:
         run: pnpm install
       - name: Perform linting
         run: pnpm lint
+        
+  build-docker:
+      name: Build Docker
+      runs-on: ubuntu-20.04
+      strategy:
+          matrix:
+              version: [ development, production, staging ]
+      needs: [ build-frontend, build-backend, lint-frontend, code-analysis ]
+      steps:
+          - uses: actions/checkout@v2
+          - name: Build the containers
+            run: docker-compose -f docker-compose.${{ matrix.version }}.yml build
```

Comment:
I don't think we need to do docker compose build for all three versions. What it now does is basically build the docker files 3 times, which is redundant. Maybe we just need to build both Dockerfiles in API and frontend, and then be done with it.